resource_customizations/elasticsearch.k8s.elastic.co/elasticsearch/health.lua

```diff
@@ -0,0 +1,34 @@
+hs = {}
+if obj.status ~= nil then
+  if obj.status.phase == "Ready" then
+    if obj.status.health == "green" then
+      hs.status = "Healthy"
+      hs.message = "Elasticsearch Cluster status is Green"
+      return hs
+    elseif obj.status.health == "yellow" then
+      hs.status = "Degraded"
+      hs.message = "Elasticsearch Cluster status is Yellow"
+      return hs
+    elseif obj.status.health == "red" then
+      hs.status = "Degraded"
+      hs.message = "Elasticsearch Cluster status is Red"
+      return hs
+    end
+  elseif obj.status.phase == "ApplyingChanges" then
+    hs.status = "Progressing" 
+    hs.message = "Elasticsearch phase is ApplyingChanges"
+    return hs
+  elseif obj.status.phase == "MigratingData" then
+    hs.status = "Progressing"
+    hs.message = "Elasticsearch phase is MigratingData"
+    return hs
+  elseif obj.status.phase == "Invalid" then
+    hs.status = "Degraded"
+    hs.message = "Elasticsearch phase is Invalid"
+    return hs
+  end
+end
+
+hs.status = "Unknown"
+hs.message = "Elasticsearch Cluster status is unknown"
```

Comment:
Do you want to suggest filing a bug? :-)

Reply:
It is intended as a catch-all. We should be able to match all cases with the current configuration so it should never trigger...unless they change something:
https://github.com/elastic/cloud-on-k8s/blob/6bb3dc7f6c0d8753a30f87ce69b7de8f0d9e0f35/pkg/apis/elasticsearch/v1/elasticsearch_types.go#L347
Do you think it could work ok like this or would you rather a different behaviour?

Reply:
Not speaking w/ a project hat (I find hats awkward). If it should never happen, then when it does happen, telling the user that they should investigate because it should never happen is probably useful.

```suggestion
hs.status = "Unknown"
hs.message = "Elasticsearch Cluster status is unknown. Ensure your ArgoCD is current and then check for/file a bug report: https://github.com/argoproj/argo-cd/issues" 
```

I generally want tools to explain to me not just their state, but what I find more important is _what should I do with this state?_

For perspective. GitHub sends me a monthly message which causes me to panic. If the message told me "you're going to get this misleading message halfway through the month telling you that you're not going to be in trouble", I'd be much happier than I am (the message is scary and makes me think I'm going to get cut off).

If the message were real and meant "you're about to get cut off, you should do X or Y in order to not get cut off", that would be better. (Right now, it says "You could do Y=allocate more money", but I don't need to, because I'm not running out.)

Reply:
This is brilliant and makes complete sense. Thank you!

Reply:
In your case, you're shipping a piece of software which is current to another piece of software. It's quite possible that the other software will change, in which case there will be a (hopefully) short period where we need to know, so we can fix it, and then a long period where people are using an outdated version of this software and they just need to update.

Worst case, their software never changes and no one sees the message :-).